src/main/java/de/bonndan/nivio/input/TemplateResolver.java

```diff
@@ -1,30 +1,44 @@
 package de.bonndan.nivio.input;
 
+import de.bonndan.nivio.ProcessingException;
 import de.bonndan.nivio.input.dto.ItemDescription;
 import de.bonndan.nivio.input.dto.LandscapeDescription;
-import de.bonndan.nivio.model.*;
-import org.springframework.util.StringUtils;
+import de.bonndan.nivio.util.URLHelper;
 
-import java.util.*;
-
-import static de.bonndan.nivio.util.SafeAssign.assignSafeIfAbsent;
+import java.net.URL;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
 
 /**
  * Responsible to apply templates to landscape items.
  * <p>
  * Resolves the items the templates are assigned to as well as dynamic endpoints of relation in templates.
  */
-public class TemplateResolver {
+public class TemplateResolver extends Resolver {
+
+    protected TemplateResolver(ProcessLog processLog) {
+        super(processLog);
+    }
 
     /**
      * Applies the template values and relations to all items the template is assigned to.
      *
-     * @param landscape           the landscape containing all(!) items. Querying happens on these items.
-     * @param templatesAndTargets The assignment can be a list of static item identifiers or dynamic queries to match
-     *                            certain items.
+     * @param landscape the landscape containing all(!) items. Querying happens on these items.
      */
-    public void processTemplates(LandscapeDescription landscape, Map<ItemDescription, List<String>> templatesAndTargets) {
-        templatesAndTargets.forEach((landscapeItem, identifiers) -> applyTemplateValues(landscapeItem, identifiers, landscape));
+    @Override
+    public void resolve(LandscapeDescription landscape) {
+
+        Map<ItemDescription, List<String>> templatesAndTargets = new HashMap<>();
+        landscape.getSourceReferences().forEach(ref -> {
+            ref.getAssignTemplates().forEach((key, identifiers) -> {
+                ItemDescription template = landscape.getTemplates().get(key);
+                template.setName(null);
```

Comment:
Why is the name set to `null`?

Reply:
Because names are more or less unique and I think it makes no sense to apply the same name to different items.

Reply:
Isn't the name then `null` by default for a template? Why set it explicitly?

Reply:
I mean, annotate it as `@Nullable`, make the getter either return `@Nullable` or  @Nonnull` and an `Optional<String>` and the caller of the getter knows what to expect.

Also, the name is annotated with `@NotEmpty` in `ItemDescription`. I think `null` is not a valid value - is it?

Reply:
ItemDescription is misused as template. And perhaps you are right: there could exist items without a name (not without identifier, however).

Reply:
I have updated the interface and added `@nullable` annotations